File: C++/Container With Most Water.cpp
#include<iostream>
#include<vector>


int maxArea(vector<int> &height)
    {
        if(height.size()==0) return 0;
        int left=0,right=height.size()-1;
        int Area=0;
        while(left<=right)
            {
                int cnt=(right-left)*min(height[left],height[right]);
                if(cnt>Area) Area=cnt;
                if(height[left]<height[right]) left++;
                else right--;
            }
        return Area;
    }

File: Python/Container-With-Most-Water.py
#!/usr/bin/env python


def maxArea(self, height):
    num = len(height)
    left = 0
    right = num - 1
    Area = (num - 1) * min(height[left], height[right])
    while left <= right:
        while left <= right and height[left] <= height[right]:
            left += 1
            Area = max(Area, (right - left + 1) * min(height[left], height[right]))
        while left < right and height[left] > height[right]:
            right -= 1
            Area = max(Area, (right - left + 1) * min(height[left], height[right]))
    Area = max(Area, (right - left) * min(height[left], height[right]))
    return Area